yeat/cli/config.py

```diff
@@ -54,3 +50,36 @@ def validate(config):
             keystr = ",".join(sorted(extrakeys))
             message = f"Ignoring unsupported configuration key(s) '{keystr}'"
             warn(message)
+
+
+class Sample:
+    def __init__(self, data):
+        sample = []
+        for fastq in data:
+            read = Path(fastq).resolve()
+            if not read.is_file():
+                raise FileNotFoundError(f"No such file: '{read}'")
+            if str(read) in sample:
+                message = f"Found duplicate read sample: '{read}'"
+                raise AssemblyConfigurationError(message)
+            else:
+                sample.append(str(read))
+        self.sample = sample
+
+    @classmethod
+    def from_json(cls, data):
+        return cls(data)
+
+
+class Assembler:
+    def __init__(self, label, algorithm, samples, extra_args=""):
+        self.label = label
+        if algorithm not in ALGORITHMS:
+            raise AssemblyConfigurationError(f"Unsupported assembly algorithm '{algorithm}'")
+        self.algorithm = algorithm
+        self.samples = samples
+        self.extra_args = extra_args
+
+    @classmethod
+    def from_json(cls, data):
+        return cls(data["label"], data["algorithm"], data["samples"], data["extra_args"])
```

Comment:
Went with an OOP route. Not sure if it made the code better or worse. The `Sample` and `Assembler` class objects are to represent portions or sections of the config. Within each object, error checking is done to ensure that the input config file has no user errors.

Example of how the OOP classes are used:
```
{
    "samples": {
        "sample1": SAMPLE_OBJ,
        "sample2": SAMPLE_OBJ,
        "sample3": SAMPLE_OBJ
    },
    "assemblers": [
        ASSEMBLER_OBJ,
        ASSEMBLER_OBJ,
        ASSEMBLER_OBJ,
        ASSEMBLER_OBJ,
        ASSEMBLER_OBJ
    ]
}
```